### ISSUE TITLE
Target child property not instantiated when mapping a derived type

### ISSUE THREAD

**Issue by GothikX** [agileobjects/AgileMapper#15]
One more I came across; tried to narrow it down some more but I'm out of time. I promise for the next one I'll take the time to understand everything in depth and make a pull request instead :)

I get a nullref exception if I have that Call there because Target is null - but only when I use the more derived class, i.e. the type of Account.PersonalDetails is TestPersonalDetails instead of PersonalDetails. If I have nothing in the Call() that would throw an exception the objects are correctly created, it seems.

Classes:

```
    public class Account
    {
        public int Id { get; set; }
        public TestPersonalDetails PersonalDetails { get; set; }

        public Account(){}
    }
    public class TestPersonalDetails : PersonalDetails
    {
        public TestPersonalDetails() {}

        public TestPersonalDetails(PersonalDetails source) : base(source)
        {
        }
    }

    public class PersonalDetails
    {
        public PersonalDetails(){}

        public PersonalDetails(PersonalDetails source)
        {
            Member = source.Member;
        }
        public string Member { get; set; }
    }

    public class AccountResponse
    {
        public string Id { get; set; }
        public AccountHolderDetails AccountHolder { get; set; }
    }

    public class AccountHolderDetails
    {
        public string Property { get; set; }
    }
```

Code that fails (m being a new mapper instance):

```
    m.WhenMapping.From<TestPersonalDetails>()
        .To<AccountHolderDetails>()
        .After.MappingEnds.Call(d =>
        {
            d.Target.Property = "asd";
        });

    m.WhenMapping.From<Account>()
        .To<AccountResponse>()
        .Map(a => a.Source.PersonalDetails)
        .To(t => t.AccountHolder);

    var source = new Account()
    {
        PersonalDetails = new TestPersonalDetails()
        {
            Member = "asd"
        }
    };

    m.Map(source).ToANew<AccountResponse>();
```

**Comment by SteveWilkes** [comment]
Interesting find! I'll sort this one out this week too :)

**Comment by SteveWilkes** [comment]
This is fixed in the latest code - release to follow :)

**Comment by SteveWilkes** [comment]
This is now fixed in v0.11.

I've also added a configuration option you might find useful because your derived types aren't all in the same assembly as their base type; you can use:

```C#
mapper.WhenMapping
    .LookForDerivedTypesIn(typeof(DerivedType1).Assembly, typeof(DerivedType2).Assembly);
```

...the mapper will check any assemblies you configure for derived types and use what it finds to construct potential mapping paths. By default it only looks in the base type's assembly.
